File: ui/src/app/services/note/note-form.ts
import {Note} from './note';


export class NoteForm {
  id: number;
  title: string;
  finished: boolean;

  setFromNote(note: Note): this {
    this.title = note.title;
    this.id = note.id;
    this.finished = note.finished;

    return this;
  }
}

File: server/src/main/java/com/bsc/notes/model/NoteForm.java
package com.bsc.notes.model;

/**
 * This class represents a model for the note creation object.
 */
public class NoteForm {
    private long id;
    private String title;
    private boolean finished;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public boolean isFinished() {
        return finished;
    }

    public void setFinished(boolean finished) {
        this.finished = finished;
    }
}
